### ISSUE TITLE
Provide URI and permalinks for all resources

### ISSUE THREAD

**Issue by Facyla** [linagora/twake-drive-legacy#405]
**Is your feature request related to a problem? Please describe.**
I cannot share unique and permanent links to shared, private or public folders, which is a must-have feature when working with other people. Email or other notifications, or browsing into shared drive or shared with me drive is not a usable alternative, because it is much longer, ambiguous (because no way to compare paths to resources), and does not help to uniquely identify a precise resource. 

**Describe the solution you'd like**
Each and every identifiable resource should have a unique, permanent URI (which translates into an URL). E.g.:
- URL_A for a folder
- URL_B for a specific file (a file i a folder)
- URL_C for the file in editing mode
- URL_D for an archived version of the file
- URL_E for the latest version of a file
- etc.
The file URI scheme can be whatever helps implementing this, and should probably be based on internal GUIDs if any ( i don't have looked at the data model yet).
Could be e.g.: 
- a given file in a folder : https://tdrive.url/file/123456 or https://tdrive.url/file/123456/latest
- a given version https://tdrive.url/file/123456/history/2
- the onlyoffice direct edit link : https://tdrive.url/file/123456/edit
- etc.

**Describe alternatives you've considered**
This is an absolute must-have, alternatives are only degraded mode options.

**Additional context**
cf. #127

**Comment by Facyla** [comment]
I'd be pleased to discuss URI-related issues and help build a consistent web HTTP API.

**Comment by shepilov** [comment]
@Facyla but you can share a permanent link to a folder, just click "..." -> Manage access -> "Copy link"

**Comment by Facyla** [comment]
Yes, i'm aware of that, but this is a public sharing link that embeds authorizations into it, and not a unique, permanent URL that would always exist without specific action or access grant. 
The point is that the current URL is not shareable without granting access at the same time., while the permalink i'm suggesting would not convey access management.
